### ISSUE TITLE
i'm getting the error while requesting the api 

### ISSUE THREAD

**Issue by Somesh-Dahatonde** [prisma/prisma#17906]
```
× Unhandled exception in handler 'postLogin'.
× Error:
  Invalid `prisma.tblusers.findUnique()` invocation in
  C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:31283:38

    31280 if (!EmailValidator.validate(eventBody.loginid)) {
    31281   throw new import_http_errors3.default.BadRequest(`Invalid loginid: ${eventBody.loginid}!`);
    31282 }
  → 31283 const user = await prisma.tblusers.findUnique(
  Query engine library for current platform "windows" could not be found.
  You incorrectly pinned it to windows

  This probably happens, because you built Prisma Client on a different platform.
  (Prisma Client looked in "C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\query_engine-windows.dll.node")

  Searched Locations:

    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\.prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\node_modules\@prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client
    /tmp/prisma-engines
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client

  You already added the platforms "windows", "linux-arm64-openssl-1.0.x" to the "generator" block
  in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
  but something went wrong. That's suboptimal.

  Please create an issue at https://github.com/prisma/prisma/issues/new
      at Kr.handleRequestError (C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30242:490)
      at Kr.handleAndLogRequestError (C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30233:16)
      at C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30517:29   
      at C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30520:95   
      at t._executeRequest (C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30752:21)
      at t._request (C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:30748:18)
      at Object.post_login (C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:31283:16)
×
  Invalid `prisma.tblusers.findUnique()` invocation in
  C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\post_login.js:31283:38

    31280 if (!EmailValidator.validate(eventBody.loginid)) {
    31281   throw new import_http_errors3.default.BadRequest(`Invalid loginid: ${eventBody.loginid}!`);
    31282 }
  → 31283 const user = await prisma.tblusers.findUnique(
  Query engine library for current platform "windows" could not be found.
  You incorrectly pinned it to windows

  This probably happens, because you built Prisma Client on a different platform.
  (Prisma Client looked in "C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module\handlers\query_engine-windows.dll.node")

  Searched Locations:

    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\.prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\node_modules\@prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\app\modules\users-module
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client
    /tmp/prisma-engines
    C:\Users\Acer\Desktop\vendorportalapi\.esbuild\.build\node_modules\.prisma\client

  You already added the platforms "windows", "linux-arm64-openssl-1.0.x" to the "generator" block
  in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
  but something went wrong. That's suboptimal.

  Please create an issue at https://github.com/prisma/prisma/issues/new
```

**Comment by mcbethio** [comment]
Hey thanks for raising an issue.

To help us reproduce please can you send us your schema, code snippets and prisma version information.

Thanks

**Comment by Somesh-Dahatonde** [comment]
ya sure

**Comment by Somesh-Dahatonde** [comment]
![image](https://user-images.githubusercontent.com/63030834/218664919-1169046d-c464-4aca-8d3d-8a23032084f4.png)
![image](https://user-images.githubusercontent.com/63030834/218665060-107a9cae-f629-48bf-bc4c-24c8b24a0f57.png)

**Comment by millsp** [comment]
Hey, we wanted to give you an update about this. We have around 10 issues that we have identified to be in the "engine not found" category. Looking at these, we learned that there are four main categories responsible for these errors:

* Your client was generated with `native` (default) on one platform, but used on a different platform at runtime.
* You pinned `binaryTargets` incorrectly, and the correct engine was therefore not included at runtime.
* A bundler/framework changed the Prisma Client but did not copy the necessary engine for the runtime.
* Deployment/tooling did not copy the engine correctly so the correct engine was not included at runtime.

These cases were previously not detected. On top of that, the error messages were confusing and were neither helpful for you or us. With that, it is likely that many of the cases above overlapped each other. So while we cannot fix a missing engine, we can detect how and why it happened. Thanks to that, we now display more useful errors to help you out of trouble. And in some cases, we will ask some input from you to learn more about your project so we can further improve.

If something is still off for you, feel free to create a new issue.
